enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/ChartTypeSelectorList/ChartTypeSelectorList.tsx

```diff
@@ -0,0 +1,93 @@
+import { useMemo } from "react";
+
+import { isNotNull } from "metabase/lib/types";
+import { Icon, Menu } from "metabase/ui";
+import visualizations from "metabase/visualizations";
+import type { Visualization } from "metabase/visualizations/types";
+import type { CardDisplayType } from "metabase-types/api";
+
+import { useQuestionVisualization } from "../../hooks/use-question-visualization";
+import { useSensibleVisualizations } from "../../hooks/use-sensible-visualizations";
+import { ToolbarButton } from "../util/ToolbarButton";
+
+export const ChartTypeSelectorList = () => {
+  const { selectedVisualization, updateQuestionVisualization } =
+    useQuestionVisualization();
+
+  const { sensibleVisualizations, nonSensibleVisualizations } =
+    useSensibleVisualizations();
+
+  const getVisualizationItems = (
+    visualizationType: CardDisplayType,
+  ): {
+    value: CardDisplayType;
+    label: Visualization["uiName"];
+    iconName: Visualization["iconName"];
+  } | null => {
+    const visualization = visualizations.get(visualizationType);
+    if (!visualization) {
+      return null;
+    }
+
+    return {
+      value: visualizationType,
+      label: visualization.uiName,
+      iconName: visualization.iconName,
+    };
+  };
+
+  const sensibleItems = useMemo(
+    () => sensibleVisualizations.map(getVisualizationItems).filter(isNotNull),
+    [sensibleVisualizations],
+  );
+  const nonsensibleItems = useMemo(
+    () =>
+      nonSensibleVisualizations.map(getVisualizationItems).filter(isNotNull),
+    [nonSensibleVisualizations],
+  );
+
+  const selectedElem = useMemo(
+    () =>
+      getVisualizationItems(selectedVisualization) ??
+      sensibleItems[0] ??
+      nonsensibleItems[0],
+    [selectedVisualization, sensibleItems, nonsensibleItems],
+  );
+
+  return (
+    <Menu position="bottom-start">
+      <Menu.Target>
+        <ToolbarButton
+          label={selectedElem.label}
+          icon={selectedElem.iconName}
+          isHighlighted={false}
+          variant="default"
+          rightIcon={<Icon ml="xs" size={10} name="chevrondown" />}
+        />
+      </Menu.Target>
+      <Menu.Dropdown h="30rem">
+        <Menu.Label>Sensible</Menu.Label>
+        {sensibleItems.map(({ iconName, label, value }, index) => (
+          <Menu.Item
+            key={`${value}/${index}`}
```

Comment:
is this because `value` can be duplicated somehow?

Reply:
yeah exactly. it would be weird but it is plausiable